osu.Game/Graphics/UserInterface/FocusedTextBox.cs

```diff
@@ -22,7 +22,10 @@ public class FocusedTextBox : OsuTextBox, IKeyBindingHandler<GlobalAction>
 
         public void TakeFocus()
         {
-            if (allowImmediateFocus) GetContainingInputManager().ChangeFocus(this);
+            if (!allowImmediateFocus)
+                return;
+
+            Schedule(() => GetContainingInputManager().ChangeFocus(this));
```

Comment:
Maybe should be allowed to run inline to avoid delaying focus transfer by a frame?

```suggestion
            Scheduler.Add(() => GetContainingInputManager().ChangeFocus(this), false);
```

Reply:
Although the same pattern was used in several other places, I guess this is fair.